### ISSUE TITLE
Sound class constructor doesn't invoke callback with error object if file not found

### ISSUE THREAD

**Issue by ashishpandey001** [zmxv/react-native-sound#305]
Hello,

It recently came under my observation that `react-native-sound` does not invoke the callback method if it is unable to find the file whose sound object needs to be created. How do you suggest implementing a check to see if the file exists within the `android/app/src/main/res/raw` directory ? This needs to be handled as the API documentation says that it has been implemented [here](https://github.com/zmxv/react-native-sound/wiki/API#constructorfilename-basepath-onerror-options) as such. Here's a sample app demonstrating the same - https://github.com/hmaster0/RNSBugExample

**Comment by acharlop** [comment]
I have the same issue when passing in a URL. If the url is invalid or the server doesn't respond the error callback is never fired.
```
new Sound(url, null, error => {
    if (error) {
        // never makes it here
        console.error('failed to load the sound', error)
        return
    }
    // loaded successfully
    // do cool stuff with audio ...
})
```

**Comment by adityasodhani** [comment]
Any fix or workaround?

**Comment by paweldomas** [comment]
There's a PR which fixes this issue:

https://github.com/zmxv/react-native-sound/pull/405

**Comment by paulmelnikow** [comment]
Fixed in #405!
